astropy/coordinates/representation.py

```diff
@@ -3059,6 +3122,39 @@ def from_representation(cls, representation, base=None):
 
         return super().from_representation(representation, base)
 
+    def transform(self, matrix, base, transformed_base):
+        """Transform differential using a 3x3 matrix in a Cartesian basis.
+
+        This returns a new differential and does not modify the original one.
+
+        Parameters
+        ----------
+        matrix : (3,3) array-like
+            A 3x3 (or stack thereof) matrix, such as a rotation matrix.
+        base : instance of ``cls.base_representation``
+            Base relative to which the differentials are defined.  If the other
+            class is a differential representation, the base will be converted
+            to its ``base_representation``.
+        transformed_base : instance of ``cls.base_representation``
+            Base relative to which the transformed differentials are defined.
+            If the other class is a differential representation, the base will
+            be converted to its ``base_representation``.
+        """
+        # the transformation matrix does not need to be a rotation matrix,
+        # so the unit-distance is not guaranteed. For speed, we check if the
+        # matrix is in O(3) and preserves lengths.
+        if np.all(is_O3(matrix)):  # remain in unit-rep
+            # TODO! implement without Cartesian intermediate step.
+            diff = super().transform(matrix, base, transformed_base)
+
+        else:  # switch to dimensional representation
+            du = (self.d_lat.unit / u.rad).decompose()  # derivative unit
```

Comment:
Here, too.  Would it make sense to `pytest.mark.parametrize` the two differentials so one tests both?